main.py

```diff
@@ -190,12 +190,18 @@ def trace_step(self, switch, entries, stored_flows):
                 'entries': entries}
 
     @classmethod
-    def do_match(cls, flow, args):
+    def do_match(cls, flow, args, table_id):
         """Match a packet against this flow (OF1.3)."""
         # pylint: disable=consider-using-dict-items
         # pylint: disable=too-many-return-statements
         if ('match' not in flow['flow']) or (len(flow['flow']['match']) == 0):
             return False
+        if 'table_id' in flow['flow']:
```

Comment:
Redundant check if `table_id` exists, `stored_flows` will guarantee the key is defined. If you can keep a `flow['flow'].get('table_id', 0)` if you wish though.